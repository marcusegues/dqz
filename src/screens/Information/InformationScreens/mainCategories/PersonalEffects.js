// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
// $FlowFixMe
import type { TFunction } from '../../../../types/generalTypes';
import { verticalScale } from '../../../../styles/Scaling';
import { AppInfoSubText } from '../../../AppInfo/subComponents/AppInfoSubText';
import { BulletText } from '../../../AppInfo/subComponents/BulletText';
import { CardRowText } from '../../../../components/QuestionAnswer/Cards/subcomponents/CardRowText';
import { SubCategoryContainer } from '../../subComponents/SubCategoryContainer';

type PersonalEffectsInnerProps = {
  t: TFunction,
};

const PersonalEffectsInner = ({ t }: PersonalEffectsInnerProps) => (
  <SubCategoryContainer title={t('personalEffectsMainText')}>
    <AppInfoSubText
      text={t('personalEffectsText1')}
      style={{ marginTop: verticalScale(10) }}
    />

    <CardRowText
      text={t('personalEffectsText2')}
      style={{ marginTop: verticalScale(20) }}
    />
    <AppInfoSubText text={t('personalEffectsText3')} />
    <BulletText text={t('personalEffectsText4')} />
    <BulletText text={t('personalEffectsText5')} />
    <CardRowText
      text={t('personalEffectsText6')}
      style={{ marginTop: verticalScale(20) }}
    />
    <AppInfoSubText text={t('personalEffectsText7')} />
  </SubCategoryContainer>
);

export const PersonalEffects = (translate(['personalEffects'])(
  PersonalEffectsInner
): ComponentType<PersonalEffectsInnerProps>);
