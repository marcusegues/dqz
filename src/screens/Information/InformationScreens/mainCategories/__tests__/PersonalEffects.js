import React from 'react';
import { shallow } from 'enzyme';
import { PersonalEffects } from '../PersonalEffects';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

describe('PersonalEffects', () => {
  test('PersonalEffects renders as expected', () => {
    const wrapper = shallow(<PersonalEffects />);
    expect(wrapper).toMatchSnapshot();
  });
  test('PersonalEffects renders as expected with language', () => {
    const wrapper = shallow(
      <PersonalEffects
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('PersonalEffects renders as expected with language with dive()', () => {
    const wrapper = shallow(
      <PersonalEffects
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de', t: 'test' }}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
