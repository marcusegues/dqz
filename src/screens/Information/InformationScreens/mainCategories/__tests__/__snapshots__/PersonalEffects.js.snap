// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalEffects PersonalEffects renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PersonalEffects PersonalEffects renders as expected with language 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    i18n={
      Object {
        "language": "de",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "i18n": Object {
        "language": "de",
      },
      "navigation": Object {
        "dispatch": [Function],
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "i18n": Object {
          "language": "de",
        },
        "navigation": Object {
          "dispatch": [Function],
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PersonalEffects PersonalEffects renders as expected with language with dive() 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonalEffectsInner
    i18n={
      Object {
        "language": "de",
        "t": "test",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <AppInfoSubText
          greyBox={false}
          style={
            Object {
              "marginTop": 19.61764705882353,
            }
          }
          text=""
        />,
        <CardRowText
          greyBox={false}
          style={
            Object {
              "marginTop": 39.23529411764706,
            }
          }
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <BulletText
          additionalText=""
          additionalTextStyle={Object {}}
          style={Object {}}
          text=""
        />,
        <BulletText
          additionalText=""
          additionalTextStyle={Object {}}
          style={Object {}}
          text=""
        />,
        <CardRowText
          greyBox={false}
          style={
            Object {
              "marginTop": 39.23529411764706,
            }
          }
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
      ],
      "title": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 19.61764705882353,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 39.23529411764706,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "additionalText": "",
          "additionalTextStyle": Object {},
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "additionalText": "",
          "additionalTextStyle": Object {},
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 39.23529411764706,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "marginTop": 19.61764705882353,
              }
            }
            text=""
          />,
          <CardRowText
            greyBox={false}
            style={
              Object {
                "marginTop": 39.23529411764706,
              }
            }
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <BulletText
            additionalText=""
            additionalTextStyle={Object {}}
            style={Object {}}
            text=""
          />,
          <BulletText
            additionalText=""
            additionalTextStyle={Object {}}
            style={Object {}}
            text=""
          />,
          <CardRowText
            greyBox={false}
            style={
              Object {
                "marginTop": 39.23529411764706,
              }
            }
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
        ],
        "title": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 19.61764705882353,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 39.23529411764706,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "additionalText": "",
            "additionalTextStyle": Object {},
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "additionalText": "",
            "additionalTextStyle": Object {},
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 39.23529411764706,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
